OpenTrons/opentrons-api
# feat(protocol-designer): wire up rename step

re AUTH-805



# Overview



Update the rename step feature to allow users to rename a step and add step notes when they click the rename button on step forms.

## Test Plan and Hands on Testing



- Go to Protocol steps and add a step
- Click the rename button and confirm that a modal appears
- Save the new step name and step notes
- Verify that the step name is renamed (the step details will be added below the command summary as the next task).
- Ensure it matches the [design](https://www.figma.com/design/WbkiUyU8VhtKz0JSuIFA45/Feature%3A-Protocol-Designer-Phase-1?node-id=11219-264190&t=1OIUHns25vPqdMDK-4)


## Changelog


- Created a new `RenameStepModal` under the `organisms` directory to display a modal for users to update the step name and step details
- Added `renameStep()` and `RenameStepAction` interface in `labware-ingred/actions/actions.ts` to handle changes
- Updated `StepFormToolbox` to handle rename button click
- Added a unit test for `RenameStepModal`
## Review requests


- I used `CHANGE_FORM_INPUT` as the type for the `RenameStepAction` interface, taking it from the old PD `ChangeFormInputAction` to handle `stepName` and `stepDetails` updates. Am I supposed to use this, or should I create a new type, add it to `labware-ingred/actions/actions.ts`, and use that instead?

## Risk assessment
